src/sema/expression/resolve_expression.rs

```diff
@@ -454,6 +462,11 @@ pub fn expression(
 
             get_int_length(&expr_type, loc, false, ns, diagnostics)?;
 
+            diagnostics.push(Diagnostic::error(
+                *loc,
+                "unary plus not permitted".to_string(),
+            ));
+
             Ok(expr)
```

Comment:
Why are you still returning `Ok` if there has been an error?

Reply:
There is no reason to stop resolving the expression/function. We want to resolve as much as possible.

Reply:
> There is no reason to stop resolving the expression/function. We want to resolve as much as possible.

You returned an error in the previous case.